tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -575,6 +576,110 @@ public:
+public:
+   using Result_t = ::TGraphAsymmErrors;
+
+private:
+   std::vector<::TGraphAsymmErrors *> fGraphAsymmErrors;
+
+public:
+   FillTGraphAsymmErrorsHelper(FillTGraphAsymmErrorsHelper &&) = default;
+   FillTGraphAsymmErrorsHelper(const FillTGraphAsymmErrorsHelper &) = delete;
+
+   // The last parameter is always false, as at the moment there is no way to propagate the parameter from the user to
+   // this method
+   FillTGraphAsymmErrorsHelper(const std::shared_ptr<::TGraphAsymmErrors> &g, const unsigned int nSlots)
+      : fGraphAsymmErrors(nSlots, nullptr)
+   {
+      fGraphAsymmErrors[0] = g.get();
+      // Initialize all other slots
+      for (unsigned int i = 1; i < nSlots; ++i) {
+         fGraphAsymmErrors[i] = new TGraphAsymmErrors(*fGraphAsymmErrors[0]);
+      }
+   }
+
+   void Initialize() {}
+   void InitTask(TTreeReader *, unsigned int) {}
+
+   // case: all types are container types
+   template <
+      typename X, typename Y, typename EXL, typename EXH, typename EYL, typename EYH,
+      std::enable_if_t<IsDataContainer<X>::value && IsDataContainer<Y>::value && IsDataContainer<EXL>::value &&
+                          IsDataContainer<EXH>::value && IsDataContainer<EYL>::value && IsDataContainer<EYH>::value,
+                       int> = 0>
+   void
+   Exec(unsigned int slot, const X &xs, const Y &ys, const EXL &exls, const EXH &exhs, const EYL &eyls, const EYH &eyhs)
+   {
+      if ((xs.size() != ys.size()) || (xs.size() != exls.size()) || (xs.size() != exhs.size()) ||
+          (xs.size() != eyls.size()) || (xs.size() != eyhs.size())) {
+         throw std::runtime_error("Cannot fill GraphAsymmErrors with values in containers of different sizes.");
+      }
+      auto thisSlotG = fGraphAsymmErrors[slot];
+      auto xsIt = std::begin(xs);
+      auto ysIt = std::begin(ys);
+      auto exlsIt = std::begin(exls);
+      auto exhsIt = std::begin(exhs);
+      auto eylsIt = std::begin(eyls);
+      auto eyhsIt = std::begin(eyhs);
+      while (xsIt != std::end(xs)) {
+         const auto n = thisSlotG->GetN(); // create only one point per iteration, but not 2
+         thisSlotG->SetPoint(n, *xsIt++, *ysIt++);
+         thisSlotG->SetPointError(n, *exlsIt++, *exhsIt++, *eylsIt++, *eyhsIt++);
+      }
+   }
+
+   // case: all types are non-container types, e.g. scalars
+   template <
+      typename X, typename Y, typename EXL, typename EXH, typename EYL, typename EYH,
+      std::enable_if_t<!IsDataContainer<X>::value && !IsDataContainer<Y>::value && !IsDataContainer<EXL>::value &&
+                          !IsDataContainer<EXH>::value && !IsDataContainer<EYL>::value && !IsDataContainer<EYH>::value,
+                       int> = 0>
+   void Exec(unsigned int slot, X x, Y y, EXL exl, EXH exh, EYL eyl, EYH eyh)
+   {
+      auto thisSlotG = fGraphAsymmErrors[slot];
+      thisSlotG->SetPoint(0, x, y);
+      thisSlotG->SetPointError(0, exl, exh, eyl, eyh);
```

Comment:
IIUC when the input is scalars, we were only ever adding one point to the graph, at index 0. I would expect a test to catch this..?

```suggestion
      const auto n = thisSlotG->GetN();
      thisSlotG->SetPoint(n, x, y);
      thisSlotG->SetPointError(n, exl, exh, eyl, eyh);
```

Reply:
I agree that I can test the index remained at 0, but I am not sure why should I introduce `const auto n = thisSlotG->GetN();`, but not derictly set the point and point error at index 0 (as I did).